src/annotations.h

```diff
@@ -6,18 +6,15 @@
 #ifndef TIMESCALEDB_ANNOTATIONS_H
 #define TIMESCALEDB_ANNOTATIONS_H
 
-/* Fall-through annotation */
-#if defined(__clang__)
-#if (__clang_major__ >= 12) || (__clang_analyzer__)
+
+/* Supported since clang 12 and GCC 7 */
+#if defined __has_attribute
+#if __has_attribute (fallthrough)
```

Comment:
Is this standard across both clang and gcc? I couldn't find good information about it so that's why I didn't use it.

Reply:
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html
https://clang.llvm.org/docs/LanguageExtensions.html#has-attribute

Reply:
I was more concern at what version this was introduced and whether one can rely on the check being there for all versions that support the attribute. It seems as if `__attribute__((fallthrough))` was added to clang 12 (I couldn't find docs for previous versions), and I guess the macro check was added prior to that or in the same release so it should be ok.